geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/Strategy.scala

```diff
@@ -201,7 +209,29 @@ object Strategy extends LazyLogging {
 trait StrategyProvider {
 
   /**
-   * Gets the estimated cost of running the query
+   * Gets the estimated cost of running the query. In general, this is the estimated
+   * number of features that will have to be scanned.
    */
-  def getCost(filter: QueryFilter, sft: SimpleFeatureType, hints: StrategyHints): Int
+  def getCost(sft: SimpleFeatureType,
+              filter: QueryFilter,
+              transform: Option[SimpleFeatureType],
+              stats: GeoMesaStats,
+              eval: CostEvaluation): Long = {
+    if (eval == CostEvaluation.Stats) {
+      statsBasedCost(sft, filter, transform, stats).getOrElse(indexBasedCost(sft, filter, transform))
+    } else if (eval == CostEvaluation.Index) {
+      indexBasedCost(sft, filter, transform)
+    } else {
+      throw new NotImplementedError(s"Unknown cost evaluation type $eval")
```

Comment:
Yeah, since it is an enumeration, I was thinking that a sealed trait might be another option.

We need an SPI to decide which components are SPI-loadable.... then the ice cream will consume itself...

Meh, this one ain't worth thinking about more.